src/state_machine.zig

```diff
@@ -386,6 +386,66 @@ pub fn StateMachineType(
             return vsr.Operation.to(StateMachine, operation);
         }
 
+        pub fn EventType(comptime operation: Operation) type {
+            return switch (operation) {
+                .pulse => void,
+                .create_accounts => Account,
+                .create_transfers => Transfer,
+                .lookup_accounts => u128,
+                .lookup_transfers => u128,
+                .get_account_transfers => AccountFilter,
+                .get_account_balances => AccountFilter,
+                .query_accounts => QueryFilter,
+                .query_transfers => QueryFilter,
+            };
+        }
+
+        pub fn ResultType(comptime operation: Operation) type {
+            return switch (operation) {
+                .pulse => void,
+                .create_accounts => CreateAccountsResult,
+                .create_transfers => CreateTransfersResult,
+                .lookup_accounts => Account,
+                .lookup_transfers => Transfer,
+                .get_account_transfers => Transfer,
+                .get_account_balances => AccountBalance,
+                .query_accounts => Account,
+                .query_transfers => Transfer,
+            };
+        }
+
+        /// Used by client code generation to make clearer APIs and for type hinting: if the Event
+        /// is a slice / list / array.
+        pub fn event_is_slice(operation: Operation) bool {
+            return switch (operation) {
+                .pulse => false,
+                .create_accounts => true,
+                .create_transfers => true,
+                .lookup_accounts => true,
+                .lookup_transfers => true,
+                .get_account_transfers => false,
+                .get_account_balances => false,
+                .query_accounts => false,
+                .query_transfers => false,
+            };
+        }
+
+        /// Used by client code generation to make clearer APIs: the name of the Event parameter,
+        /// when used as a variable.
+        pub fn event_name(operation: Operation) []const u8 {
+            return switch (operation) {
+                .pulse => "none",
+                .create_accounts => "accounts",
+                .create_transfers => "transfers",
+                .lookup_accounts => "accounts",
+                .lookup_transfers => "transfers",
+                .get_account_transfers => "filter",
+                .get_account_balances => "filter",
+                .query_accounts => "query_filter",
+                .query_transfers => "query_filter",
+            };
+        }
+
```

Comment:
To add some context around these - the ideal goal would be for a state machine to define everything it exposes, in a way that's both checked at compile time (so you can't forget something off...) and makes it possible for a client bindings generator to generate bindings entirely from that description.

The current Python generator still relies on hardcoded types, `mappings_state_machine`, so it's not quite there yet.

(as to the how, we could hoist the execution logic of `commit()` up into VSR, so that calls into specific functions in the state machine, based on what it exposed. This would also give better type safety than `input: []align(16) const u8`)

Reply:
But... Out of scope for this PR!

Reply:
I wonder if we should extract meta information out of state machine, and include it in tigerbeetle.zig? So that python codegen and state machine depend on tigerbeetle.zig, but python doesn't depend on the state machine directly. 

The fact that we instantiate IO in our python bindings generator is a smell --- something's not factored quite right. But it's not clear _what_ is the right factoring, so definitely for the follow up!